### PR TITLE
Add error system

### PR DESCRIPTION
**Pull request by Andarist**
Prod builds went down from 9522 to 8892 bytes (min + gzipped).

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -99,6 +100,7 @@
     "ramda": "^0.25.0",
     "rollup": "^0.64.1",
     "rollup-plugin-babel": "^3.0.7",
+    "rollup-plugin-node-resolve": "^3.3.0",
```
Comment (TrySound) [inline package.json]: Do you think it's worth to add another plugin just to resolve `index.js`?

**Comment by bhough** [comment]
@Andarist Thank you for kicking this off. I'm taking a bit of a breather from V2 before diving in to V3 work, but Ill aim to get some initial comments in next week.

Also, I updated this to be agains the `version-3` branch since this is currently on the roadmap for that release (#354)

**Comment by Andarist** [comment]
Take your time, this doesnt have to wait till v3 though - what I mean is it shouldnt be a breaking change.

**Comment by bhough** [comment]
Pulled this code into #398. Thanks for the contribution @Andarist!

**Pull request closed without merging**